pySDC/implementations/problem_classes/DiscontinuousTestODE.py

```diff
@@ -0,0 +1,216 @@
+import numpy as np
+
+from pySDC.core.Errors import ParameterError, ProblemError
+from pySDC.core.Problem import ptype
+from pySDC.implementations.datatype_classes.mesh import mesh
+
+
+class DiscontinuousTestODE(ptype):
+    r"""
+    This class implements a very simple test case of a ordinary differential equation consisting of one discrete event. The dynamics of
+    the solution changes when the state function :math:`h(u) := u - 5` changes the sign. The problem is defined by:
+
+    if :math:`u - 5 < 0:`
+
+        .. math::
+            \fra{d u}{dt} = u
+
+    else:
+
+        .. math::
+            \frac{d u}{dt} = \frac{4}{t^*},
+
+    where :math:`t^* = \log(5) \approx 1.6094379`. For :math:`h(u) < 0`, i.e., :math:`t \leq t^*` the exact solution is
+    :math:`u(t) = exp(t)`; for :math:`h(u) \geq 0`, i.e., :math:`t \geq t^*` the exact solution is :math:`u(t) = \frac{4 t}{t^*} + 1`.
+
+    Attributes
+    ----------
+    t_switch_exact : float
+        Exact event time with :math:`t^* = \log(5)`.
+    t_switch: float
+        Time point of the discrete event found by switch estimation.
+    nswitches: int
+        Number of switches found by switch estimation.
+    newton_itercount: int
+        Counts the number of Newton iterations.
+    newton_ncalls: int
+        Counts the number of how often Newton is called in the simulation of the problem.
+    """
+
+    dtype_u = mesh
+    dtype_f = mesh
+
+    def __init__(self, nvars=1, newton_maxiter=200, newton_tol=1e-8):
+        """Initialization routine"""
+
+        super().__init__(init=(nvars, None, np.dtype('float64')))
+        self._makeAttributeAndRegister('nvars', 'newton_maxiter', 'newton_tol', localVars=locals(), readOnly=True)
+
+        if self.nvars != 1:
```

Comment:
Don't put `nvars` as argument, if it's only value can be one. But you can let it as read-only parameter so it can be used later fo the data initialization (but since it's stored already in `init`, I'm not sure that you need it as problem parameter)